doc/src/parts/user_guide.rst

```diff
@@ -28,5 +29,427 @@ can be divided into the following steps:
-.. TODO: link
-The full code snippets can be found under /examples. 
+
+Using kafe2go
+-------------
+Using *kafe2go* is the simplest way of performing a fit. Here all the necessary
+information like data and uncertainties is specified in the *YAML* data format.
+To perform the fit, simply run
+
+.. code-block:: bash
+
+    kafe2go path/to/fit.yml
+
+Using Python
+------------
+When using *kafe2* via a *Python* script a fine control of the fitting- and
+plotting-procedure is possible. For using *kafe2* inside a *Python* script, import the required
+*kafe2* modules:
+
+.. code-block:: python
+
+    from kafe2 import XYFit, Plot
+
+If a code example contains a line similar to :python:`data = XYContainer.from_file("data.yml")`
+the according dataset can be found in the corresponding examples folder. Those are located inside the
+installation directory of *kafe2*. Additionally the examples can be found on
+`GitHub <https://github.com/dsavoiu/kafe2/tree/master/examples>`_.
+
+Example 1: Line Fit
+===================
+The first example is the simplest use of a fitting framework, performing a line fit.
+A linear function of the form :math:`f(x;a, b) = a x + b` is made to align with
+a series of xy data points that have some uncertainty along the x-axis
+and the y-axis.
+This example demonstrates how to perform such a line fit in kafe2 and
+how to extract the results.
+
+.. figure:: ../_static/img/001_line_fit.png
+    :alt: Plot of a line fit performed with kafe2.
+
+kafe2go
+-------
+To run this example, open a text editor and save the following file contents
+as a *YAML* file named :file:`line_fit.yml`.
+
+.. literalinclude:: ../../../examples/001_line_fit/line_fit.yml
+
+Then open a terminal, navigate to the directory where the file is
+located and run
+
+.. code-block:: bash
+
+    kafe2go line_fit.yml
+
+Python
+------
+The same fit can also be performed by using a *Python* script.
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: python code
+
+    .. literalinclude:: ../../../examples/001_line_fit/line_fit.py
+        :lines: 15-
+
+
+Example 2: Model Functions
+==========================
+
+In experimental physics a line fit will only suffice for a small number
+of applications. In most cases you will need a more complex model function
+with more parameters to accurately model physical reality.
+This example demonstrates how to specify arbitrary model functions for
+a kafe2 fit.
+When a different function has to be fitted, those functions need to be defined either in the
+*YAML* file or the *Python* script.
+
+.. figure:: ../_static/img/002_model_comparison.png
+    :alt: Comparison of a line and exponential fit with kafe2.
+
+The expectation value of :math:`\chi^2/\mathrm{ndf}` is 1. Thus the exponential function
+more accurately describes the dataset than a line function.
+
+An exponential function is a **non linear** function! Therefore the :math:`\chi^2` can have a non
```

Comment:
Yes, it should be *profile* `chi2` in this case.

I also think the sentence "An exponential function is a **non linear** function!" needs to specify what a "nonlinear" model is in this context.

As we discused a few times before, the term "linear model" here refers to a model that is a linear function *of the parameters* **p**, not the independent variable *x*. In this sense, a line (i.e. *ax+b*) is a linear model, but so is a parabola (or any polynomial for that matter). An exponential is not a linear model in either sense.

Since example 2 is comparing a *line fit* to an *exponential*, beginners might think that the *linearity* in this case refers to the *x* dependence. This is something we should definitely avoid!

Maybe it should be changed to something making the difference more clear, like "An exponential function is **not** a linear function of its parameters `A0` and `tau`. Therefore [...]".

Reply:
Do you think this is good?
```
An exponential function is a **non-linear** function of it's parameters!
**Non-linear** refers to the linearity of the parameters.
So any parabola or polynomial like :math:`ax^2+bx+c` is a linear function of the parameters
:math:`a`, :math:`b` and :math:`c`.
So an exponential function :math:`A_0 e^{(x/x_0)}` is **non-linear** in its parameter :math:`x_0`.
Thus the profile :math:`\chi^2` can have a non parabolic shape.
```

Reply:
This does a better job at explaining what the linearity refers to, bu tnow the sentence about the parabola might be a source of confusion, because you talk about the chi2 having a parabolic shape in the next paragraph.

But it's an improvement in any case. We can discuss refining the exact formulation at some point in the future, but for now it's OK.